### PR TITLE
Add support for different meter types.

### PR DESCRIPTION
**Pull request by jonathan-dilorenzo**
From a new branch because I couldn't figure out how to update PR #473. See that PR for more information.

### CODE REVIEW

**Review by smolkaj — APPROVED** [review]
This looks good on my end. Can we get a second review, perhaps from @chrispsommers or @jafingerhut?

File: docs/v1/P4Runtime-Spec.mdk
Comment (jonathan-dilorenzo) [reply docs/v1/P4Runtime-Spec.mdk]: Done!

File: docs/v1/P4Runtime-Spec.mdk
Comment (chrispsommers) [inline docs/v1/P4Runtime-Spec.mdk:1765]: Is there a PR for the P4-16 Language Spec to reserve these new annotations? If so, it'd be nice to provide to cross-references in each PR to the other. If not, should this be initiated before merging this?

Comment (chrispsommers) [reply docs/v1/P4Runtime-Spec.mdk:1765]: Also, what are the implications for the various p4c backends? I can't recall if any platforms already support this and you're just extending P4RT to accommodate it. Thanks.

Comment (jonathan-dilorenzo) [reply docs/v1/P4Runtime-Spec.mdk:1765]: Good questions!

> Is there a PR for the P4-16 Language Spec to reserve these new annotations?

There is no such PR. We haven't been adding P4Runtime annotations to the P4 spec (see https://p4.org/p4-spec/docs/p4-16-working-draft.html#sec-p4-reserved-annotations), but maybe we should... Or maybe we should include a reference from the relevant appendix in the P4 Spec to the P4Runtime annotations and note that other P4-related standards may pre-define their own standardized annotations (which ideally shouldn't overlap)?

WDYT? Regardless, wouldn't block merging this PR on the decision.

> Also, what are the implications for the various p4c backends?

So, I think the implication would be basically precisely what the PR says. They should reject entries with invalid meter configs and create the related meters. They should also be able to say that they don't support a particular type though... Maybe I should add something to that effect to the PR?

Comment (chrispsommers) [reply docs/v1/P4Runtime-Spec.mdk:1765]: Hi Jonathan, thanks for the responses.
Re: Reeserved amnotations - thanks for refreshing my memory, we don't cite P4RT annos there, it's left as an exercise to the reader although somehow I thought we had a pointer to the P4RT repo so folks could check; guess not.

Re: Backends, I'm still unclear how you can use this change - is there a backend which can handle it?

Comment (smolkaj) [reply docs/v1/P4Runtime-Spec.mdk:1765]: > Re: Backends, I'm still unclear how you can use this change - is there a backend which can handle it?

Our "backend" handles it :)
But note that we don't actually have our own p4c backend. We just use the p4info backend and consume that.

Comment (smolkaj) [reply docs/v1/P4Runtime-Spec.mdk:1765]: Also, I really like Jonathan's suggestion to link from the P4 spec to the P4Runtime annotatons (but agree this PR should not be blocked on that).

Comment (chrispsommers) [reply docs/v1/P4Runtime-Spec.mdk:1765]: > > Re: Backends, I'm still unclear how you can use this change - is there a backend which can handle it?
> 
> Our "backend" handles it :) But note that we don't actually have our own p4c backend. We just use the p4info backend and consume that.

Ahh, OK, makes sense. Thanks!

**Review by chrispsommers — APPROVED** [review]
LGTM

**Pull request merged**
